src/components/icon-button/icon-button.component.tsx

```diff
@@ -21,26 +22,47 @@ export interface IconButtonProps extends SpaceProps {
   /** Callback triggered on mouse leave */
   onMouseLeave?: (ev: React.MouseEvent<HTMLButtonElement>) => void;
   /** Action callback */
```

Comment:
```suggestion
  /** [DEPRECATED - use `onClick` instead] Action callback */
```
It might be worth additionally amending the prop's description so the deprecation is visible in storybook and via intellisense